src/mjolnir/osmway.cc

```diff
@@ -173,16 +181,20 @@ OSMWay::GetNames(const std::string& ref, const UniqueNames& name_offset_map, uin
 std::vector<std::string> OSMWay::GetTaggedNames(const UniqueNames& name_offset_map) const {
 
   std::vector<std::string> names;
-  std::vector<std::string> tokens;
 
   if (tunnel_name_index_ != 0) {
     // tunnel names
-    tokens = GetTagTokens(name_offset_map.name(tunnel_name_index_));
+    auto tokens = GetTagTokens(name_offset_map.name(tunnel_name_index_));
     for (const auto& t : tokens) {
       names.emplace_back(std::to_string(static_cast<uint8_t>(TaggedName::kTunnel)) + t);
     }
   }
 
+  if (z_level_ != 0) {
+    names.emplace_back(std::to_string(static_cast<uint8_t>(TaggedName::kZLevel)) +
```

Comment:
this is on an other PR too #3132 . i think the original implementation must have screwed this up but i didnt go check the logic yet. we werent supposed to `to_string` the enum value, that doesnt make any sense, because that wont fit in a single byte! we want to cast it to a char and use that... if this is the case i think we royally screwed up this whole idea in the original implementation..

Reply:
and yeah its screwed up: https://github.com/valhalla/valhalla/pull/2608/files#r687122174

Reply:
issue is here, *looks* like we can trivially fix it: https://github.com/valhalla/valhalla/issues/3262